src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -890,15 +893,24 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
             lockFileFormat.Write(projectLockFilePath, lockFile);
         }
 
-        private void AddRemoteProvidersFromSources(List<IWalkProvider> remoteProviders, List<PackageSource> effectiveSources, SummaryContext summary)
+        private void AddRemoteProvidersFromSources(List<IWalkProvider> remoteProviders,
+                                                   List<PackageSource> effectiveSources,
+                                                   ConcurrentDictionary<PackageSource, Restore.NuGet.IPackageFeed> packageFeeds,
+                                                   SummaryContext summary)
         {
             foreach (var source in effectiveSources)
             {
-                var feed = PackageSourceUtils.CreatePackageFeed(
-                    source,
-                    FeedOptions.NoCache,
-                    FeedOptions.IgnoreFailedSources,
-                    Reports);
+                Restore.NuGet.IPackageFeed feed;
+                if (!packageFeeds.TryGetValue(source, out feed))
+                {
+                    feed = PackageSourceUtils.CreatePackageFeed(
```

Comment:
After a thread creates a feed, it may find another thread already creates the same thing and adds to the dictionary. It is nice to use `packageFeeds.GetOrAdd` to avoid this case.